### DISCUSSION TITLE
How can we use the tags in Azure Devops to decide which build is production

### DISCUSSION THREAD

**Discussion by Stepness** [apache/devlake#7955] [General]
We would like to use the tags on Azure Devops to select production builds instead of looking at jobs name to retrieve DORA data.
How can we do that?

The tag property in here https://github.com/apache/incubator-devlake/blob/main/backend/plugins/azuredevops_go/models/build.go

**Comment by klesh** [comment]
I don't think that is supported at the moment.

**Reply by Stepness** [reply]
Is there a way to modify the deployment property extracted to check the tags, or to change the dashboard to do that?

**Reply by ilgiampi** [reply]
Hi @klesh, is there any plan to support this?
If I'm not mistaken, only the ADO pipeline name or the job name can be used to infer a deployment at the moment.
With dozens of pipelines already implemented, a more flexible way to identify a deployment can significantly help.
Also, being allowed to use the stage name could simplify the configuration.

**Reply by klesh** [reply]
@Stepness I believe the most practical approach would be to modify the existing code. Deployments play a crucial role in the DORA plugin's calculations, so alterting the dashboard itself might not yield desired results.
@ilgiampi AFAIK there isn't a plan for the matter. However, you are welcome to file a Feature Request. We are open to improvements, and if anyone is interested in working on it, we'd be happy to collaborate.

**Comment by Stepness** [comment]
A feature request has been opened #7956
